### PR TITLE
docs(hx-switch): Switch documentation

### PR DESCRIPTION
**Pull request by lalithkarikelli**
## Description

`<hx-switch>`

### What are the relevant story cards/tickets? Any additional PRs or other references?

Jira: SURF2072

## Before you request a review for this PR:

- [x] For UI changes, did you manually test in recent versions of modern browsers (Chrome, Firefox, and Safari)?
- [x] For UI changes, did you manually test in IE11 and legacy Edge?
- [ ] Did you add component tests for any new code?
- [ ] Did you run the component unit tests via `yarn test` to ensure all tests passed?
- [ ] Did you include a screenshot of the component tests?
- [ ] If you changed/added functionality, did you update the demo page and documentation?
- [ ] If needed, did you add or modify the demo test page to test the changed/added functionality?
- [x] Did you assign reviewers?
- [x] In Jira, have you linked to this PR on the ticket(s)?

### CODE REVIEW

File: docs/components/switch/index.html [hunk 1]
```diff
@@ -15,6 +15,99 @@
...
+
+  </section>
+
+  <!-- end of demo yes/no label example section for switch -->
+
+  <!-- start of demo disabled example section for switch -->
+
   <section>
     <header>
       <h2 id="basic-checkbox">Switch</h2>
```
Comment (100stacks) [inline docs/components/switch/index.html:113]: Please update:

```html
<h2 id="disabled-switch">Disabled Switch</h2>
```

File: docs/components/switch/index.html [hunk 2]
```diff
@@ -15,6 +15,99 @@
   {% endblock %}
 
   {% block content %}
+
+  <!-- start of demo no label example section for switch -->
+
+  <section>
+    <header>
+      <h2 id="basic-checkbox">Switch with no label</h2>
```
Comment (100stacks) [inline docs/components/switch/index.html:23]: Please update:

```html
<h2 id="default-switch">Default Switch (no labels)</h2>
```

File: docs/components/switch/index.html [hunk 3]
```diff
@@ -15,6 +15,99 @@
...
+      <p></p>
```
Comment (100stacks) [inline docs/components/switch/index.html:24]: Please remove empty `<p>` element.

File: docs/components/switch/index.html [hunk 4]
```diff
@@ -15,6 +15,99 @@
...
+
+  </section>
+
+  <!-- end of demo no label example section for switch -->  
+
+  <!-- start of demo on/off label example section for switch -->
+
+  <section>
+    <header>
+      <h2 id="basic-checkbox">Switch with on/off label</h2>
```
Comment (100stacks) [inline docs/components/switch/index.html:51]: Please update:

```html
<h2 id="on-switch">Switch with on/off labels</h2>
```

File: docs/components/switch/index.html [hunk 5]
```diff
@@ -15,6 +15,99 @@
...
+      <p></p>
```
Comment (100stacks) [inline docs/components/switch/index.html:52]: Please remove empty `<p>` element.

File: docs/components/switch/index.html [hunk 6]
```diff
@@ -15,6 +15,99 @@
...
+    </header>
+    <p class="hxSubdued hxSubBody">
+      <hx-icon type="exclamation-triangle"></hx-icon>
+      The label can be overridden to provide a different label. The character width is limited to 2-3 characters.
+    </p>
```
Comment (100stacks) [inline docs/components/switch/index.html:57]: Please update to:

```html
<p class="hxSubdued hxSubBody">
  <hx-icon type="exclamation-triangle"></hx-icon>
  The label can be overridden to provide a different label (for i18n support).
  The character width is limited to 2-3 characters.
</p>
```

File: docs/components/switch/index.html [hunk 7]
```diff
@@ -15,6 +15,99 @@
...
+
+  <section>
+    <header>
+      <h2 id="basic-checkbox">Switch with yes/no label</h2>
+      <p></p>
+    </header>
+    <p class="hxSubdued hxSubBody">
+      <hx-icon type="exclamation-triangle"></hx-icon>
+      The label can be overridden to provide a different label. The character width is limited to 2-3 characters.
+    </p>
```
Comment (100stacks) [inline docs/components/switch/index.html:88]: Please update to:

```html
<p class="hxSubdued hxSubBody">
  <hx-icon type="exclamation-triangle"></hx-icon>
  The label can be overridden to provide a different label (for i18n support).
  The character width is limited to 2-3 characters.
</p>
```

File: docs/components/switch/index.html [hunk 8]
```diff
@@ -15,6 +15,99 @@
...
+
+  </section>
+
+  <!-- end of demo on/off label example section for switch -->
+
+  <!-- start of demo yes/no label example section for switch -->
+
+  <section>
+    <header>
+      <h2 id="basic-checkbox">Switch with yes/no label</h2>
```
Comment (100stacks) [inline docs/components/switch/index.html:82]: Please update:

```html
<h2 id="yes-switch">Switch with yes/no Labels</h2>
```

File: docs/components/switch/index.html [hunk 9]
```diff
@@ -15,6 +15,99 @@
...
+      <p></p>
```
Comment (100stacks) [inline docs/components/switch/index.html:83]: Please remove empty `<p>` element.

File: docs/components/switch/index.html [hunk 10]
```diff
@@ -56,6 +149,8 @@ <h2 id="basic-checkbox">Switch</h2>
 
   </section>
 
+  <!-- end of demo disabled example section for switch -->
+
   <!-- start of demo error example section for switch -->
 
   <section>
```
Comment (100stacks) [inline docs/components/switch/index.html:156]: Please update:

```html
<header>
  <h2 id="error-switch">Error Switch</h2>
</header>
```

File: docs/components/switch/index.html [hunk 11]
```diff
@@ -98,4 +193,4 @@ <h2 id="basic-checkbox">Error Switch</h2>
 
   </section>
   <!-- end of demo error example section for switch -->
-  {% endblock %}
+  {% endblock %}
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply docs/components/switch/index.html:194]: Changes have been done, please review.

File: docs/components/switch/index.html (hunk 1 above)
Comment (lalithkarikelli) [reply docs/components/switch/index.html]: Changes have been done, please review.

File: docs/components/switch/index.html (hunk 2 above)
Comment (lalithkarikelli) [reply docs/components/switch/index.html]: Changes have been done, please review.

File: docs/components/switch/index.html (hunk 3 above)
Comment (lalithkarikelli) [reply docs/components/switch/index.html]: Changes have been done, please review.

File: docs/components/switch/index.html (hunk 4 above)
Comment (lalithkarikelli) [reply docs/components/switch/index.html]: Changes have been done, please review.

File: docs/components/switch/index.html (hunk 5 above)
Comment (lalithkarikelli) [reply docs/components/switch/index.html]: Changes have been done, please review.

File: docs/components/switch/index.html (hunk 6 above)
Comment (lalithkarikelli) [reply docs/components/switch/index.html]: Changes have been done, please review.

File: docs/components/switch/index.html (hunk 7 above)
Comment (lalithkarikelli) [reply docs/components/switch/index.html]: Changes have been done, please review.

File: docs/components/switch/index.html (hunk 8 above)
Comment (lalithkarikelli) [reply docs/components/switch/index.html]: Changes have been done, please review.

File: docs/components/switch/index.html (hunk 9 above)
Comment (lalithkarikelli) [reply docs/components/switch/index.html]: Changes have been done, please review.

File: docs/components/switch/index.html (hunk 10 above)
Comment (lalithkarikelli) [reply docs/components/switch/index.html:154]: Changes have been done, please review.

File: docs/_data/nav.json5 [hunk 12]
```diff
@@ -105,6 +105,7 @@
             { label: '<hx-select-control>', path: 'hx-select-control' },
             { label: '<hx-status>', path: 'hx-status' },
             { label: '<hx-switch>', path: 'hx-switch' },
+            { label: '<hx-switch-control>', path: 'hx-switch-control'},
```
Comment (100stacks) [inline docs/_data/nav.json5:108]: Added missing `<hx-switch-control>` documentation.

File: docs/components/switch/error-switch-demo.js [hunk 13]
```diff
@@ -1,8 +1,8 @@
 import Util from '../../_util';
 
-if (document.getElementById('vue-errorSwitchDemo')) {
+if (document.getElementById('vue-switchErrorDemo')) {
```
Comment (100stacks) [inline docs/components/switch/error-switch-demo.js:3]: Changed to make naming convention more consistent.

File: docs/components/switch/index.html [hunk 14]
```diff
@@ -3,6 +3,7 @@
 minver: 0.24.0
 also:
   elements/hx-switch: <hx-switch>
+  elements/hx-switch-control>: <hx-switch-control>
```
Comment (100stacks) [inline docs/components/switch/index.html:6]: Added missing document for `<hx-switch-control>`.

File: docs/components/switch/index.html [hunk 15]
```diff
@@ -15,13 +16,104 @@
...
   {% block content %}
+
+  <!-- start of demo no label example section for switch -->
+
+  <section>
+    <header>
+      <h2 id="default-switch">Default Switch (no labels)</h2>
+    </header>
+
+    <div class="example" id="vue-switchNoneDemo" v-cloak>
```
Comment (100stacks) [inline docs/components/switch/index.html:27]: Changed to make naming convention more consistent.

File: docs/elements/hx-switch/index.html [hunk 16]
```diff
@@ -1,41 +1,34 @@
...
-  elements/hx-switch: <hx-switch>
+  elements/hx-switch-control: <hx-switch-control>
 ---
 {% extends 'element.njk' %}
 {% import '_utils.njk' as utils %}
 {% block page_header %}
   <p>
-    The <code>{{page.title}}</code> block element is a container that provides
-    behavior to augment native browser validation capabilities.  It is used to
-    build a <a href="components/switch">Switch</a> component.
```
Comment (100stacks) [inline docs/elements/hx-switch/index.html:15]: Update to reflect `<hx-switch>` documentation.

File: src/elements/hx-switch-control/index.js [hunk 17]
```diff
@@ -14,8 +14,21 @@ export class HXSwitchControlElement extends HXFormControlElement {
         return 'hx-switch-control';
     }
 
+    $onCreate () {
+        this._onClick = this._onClick.bind(this);
+    }
+
+    $onConnect () {
+        this.controlElement.addEventListener('click', this._onClick);
```
Comment (100stacks) [inline src/elements/hx-switch-control/index.js:22]: Added a `click` event to `toggle` the Switch Component.

File: src/elements/hx-switch/index.js [hunk 18]
```diff
@@ -67,6 +68,20 @@ export class HXSwitchElement extends HXFormControlElement {
...
+    get toggled () {
+        return this.hasAttribute('toggled');
+    }
+    set toggled (isToggled) {
+        if (isToggled) {
+            this.setAttribute('toggled', '');
+        } else {
+            this.removeAttribute('toggled');
+        }
+    }
```
Comment (100stacks) [inline src/elements/hx-switch/index.js:83]: Added a `toggled` attribute to provide consumers an option to check Switch Component state.

File: src/scss/components/switch/_index.scss [hunk 19]
```diff
@@ -109,13 +98,6 @@ hx-switch-control > input[type="checkbox"]:checked ~ label {
   }
 }
 
-.switch {
-  position: relative;
-  display: inline-block;
-  width: 4.25rem;
-  height: 2.125rem;
-}
-
```
Comment (100stacks) [inline src/scss/components/switch/_index.scss:118]: Moved this styling into `hx-switch-control`.  This should allow implementors to use:

```js
<hx-switch-control> // removed class="switch"
  <input type="checkbox" id="switchYesNoTest" />
  <label for="switchYesNoTest">
    <hx-switch
      onlabel="yes"
      offlabel="no"
      aria-labelledby="switchYesNoTest">
    </hx-switch>
  </label>
</hx-switch-control>
```

**Review by BY00565233 — APPROVED** [review]
DEV LGTM

**Review by 100stacks — APPROVED** [review]
Dev LGTM

**Pull request merged**